zebra-script/src/lib.rs

```diff
@@ -1,11 +1,198 @@
+            script_ptr,
+            script_len as u32,
+            amount,
+            tx_to_ptr,
+            tx_to_len as u32,
+            n_in,
+            #[cfg(not(windows))]
+            flags,
+            #[cfg(windows)]
+            flags.try_into().expect("why bindgen whyyy"),
+            consensus_branch_id,
+            &mut err,
+        )
+    };
+
+    if ret == 1 {
+        Ok(())
+    } else {
+        Err(Error::from(err))
+    }
+}
+
+/// Verify a script within a transaction given the corresponding
+/// `TransparentOutput` it is spending and the `ConsensusBranchId` of the block
+/// containing the transaction.
+///
+/// # Details
+///
+/// input index corresponds to the index of the `TransparentInput` which in
+/// `transaction` used to identify the `previous_output`
+pub fn is_valid(
+    transaction: Arc<Transaction>,
+    branch_id: ConsensusBranchId,
+    (input_index, previous_output): (u32, TransparentOutput),
+) -> Result<(), Error> {
+    assert!((input_index as usize) < transaction.inputs().len());
+
+    let tx_to = transaction
+        .zcash_serialize_to_vec()
+        .expect("serialization into a vec is infallible");
+
+    let TransparentOutput { value, lock_script } = previous_output;
+
+    verify_script(
+        &lock_script.0,
+        value.into(),
+        &tx_to,
+        input_index as _,
+        branch_id.into(),
+    )?;
+
+    Ok(())
+}
+
 #[cfg(test)]
 mod tests {
+    use super::*;
+    use hex::FromHex;
+    use std::convert::TryInto;
+    use std::sync::Arc;
+    use zebra_chain::{serialization::ZcashDeserializeInto, types::Script};
+    use zebra_test::prelude::*;
+
+    lazy_static::lazy_static! {
+        pub static ref SCRIPT_PUBKEY: Vec<u8> = <Vec<u8>>::from_hex("76a914f47cac1e6fec195c055994e8064ffccce0044dd788ac").unwrap();
+        pub static ref SCRIPT_TX: Vec<u8> = <Vec<u8>>::from_hex("0400008085202f8901fcaf44919d4a17f6181a02a7ebe0420be6f7dad1ef86755b81d5a9567456653c010000006a473044022035224ed7276e61affd53315eca059c92876bc2df61d84277cafd7af61d4dbf4002203ed72ea497a9f6b38eb29df08e830d99e32377edb8a574b8a289024f0241d7c40121031f54b095eae066d96b2557c1f99e40e967978a5fd117465dbec0986ca74201a6feffffff020050d6dc0100000017a9141b8a9bda4b62cd0d0582b55455d0778c86f8628f870d03c812030000001976a914e4ff5512ffafe9287992a1cd177ca6e408e0300388ac62070d0095070d000000000000000000000000").expect("Block bytes are in valid hex representation");
+    }
+
+    #[test]
+    fn verify_valid_script_parsed() -> Result<()> {
+        zebra_test::init();
+
+        let transaction =
+            SCRIPT_TX.zcash_deserialize_into::<Arc<zebra_chain::transaction::Transaction>>()?;
+        let coin = u64::pow(10, 8);
+        let amount = 212 * coin;
+        let output = TransparentOutput {
+            value: amount.try_into()?,
+            lock_script: Script(SCRIPT_PUBKEY.clone()),
+        };
+        let input_index = 0;
+        let branch_id = ConsensusBranchId::BLOSSOM;
```

Comment:
Do we need to add constants?

```suggestion
        let branch_id = NetworkUpgrade::Blossom.branch_id();
```

Reply:
Seems reasonable, but lets leave that for a different PR

Reply:
related to #909

Reply:
Reverted by #926.